src/main/scala/ai/eto/rikai/sql/spark/execution/ModelCommand.scala

```diff
@@ -20,17 +20,26 @@ import ai.eto.rikai.sql.model.Catalog
 import com.thoughtworks.enableIf
 import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.execution.command.RunnableCommand
 import org.apache.spark.sql.types.StringType
-import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 
 trait ModelCommand extends RunnableCommand {
   override final def children: Seq[LogicalPlan] = Nil
 
-  @enableIf(scala.util.Properties.versionNumberString.compareTo("2.12.15") >= 0)
+  @enableIf(c =>
+    c.classPath.exists(
+      _.getPath.matches(".*spark-catalyst_2\\.\\d+-3\\.2\\..*")
+    )
+  )
   override final def mapChildren(f: LogicalPlan => LogicalPlan): LogicalPlan =
     this.asInstanceOf[LogicalPlan]
-  @enableIf(scala.util.Properties.versionNumberString.compareTo("2.12.15") >= 0)
+
+  @enableIf(c =>
+    c.classPath.exists(
+      _.getPath.matches(".*spark-catalyst_2\\.\\d+-3\\.2\\..*")
+    )
+  )
```

Comment:
nice,
what will we do for Spark 3.3 then ;)

Reply:
We can use `_.getPath.matches(".*spark-catalyst_2\\.\\d+-3\\.(2|3)\\..*")` or similar things, `regex` is very powerful, just will become less readable as it become more powerful.